design/src/rfcs/rfc0027_new_home_for_cred_cache_and_builder_api_consistency.md

```diff
@@ -0,0 +1,182 @@
+<p align="center">
+  <img width="800" alt="Option A" src="https://user-images.githubusercontent.com/15333866/206587781-6eca3662-5096-408d-a435-d4023929e727.png">
+</p>
+
+- :+1: `aws-XXX::sdk_config::Builder` and a service client `config::Builder` can create a `SharedCredentialsCache` with a concrete type of credentials cache.
+- :+1: It avoids cyclic crate dependencies.
+- :-1: There is one more crate to maintain and version.
+- :-1: `aws-XXX::sdk_config` sitting together with the types and traits for credentials provider & caching may not give us a coherent mental model for the `aws-XXX` crate, making it difficult to choose the right name for `XXX`.
+
+#### Option B: Moving `aws_types::sdk_config` to yet another new crate
+
+A result of this arrangement will give us the following module interdependencies:
+<p align="center">
+  <img width="800" alt="Option B" src="https://user-images.githubusercontent.com/15333866/206587806-fe720118-e3d4-4dc9-9b33-5f65fc1a445c.png">
+</p>
+
+- :+1: `aws-XXX::sdk_config::Builder` and a service client `config::Builder` can create a `SharedCredentialsCache` with a concrete type of credentials cache.
+- :+1: It avoids cyclic crate dependencies.
+- :+1: The scope of `aws-XXX` is more focused, concerned primarily with credentials.
+- :-1: There are two more crates to maintain and version.
+- :-1: :-1: The scope of the new crate for `aws_types::sdk_config` seems too narrow and therefore having a separate crate may not be worth it.
+
+
+Rejected Ideas
+--------------
+### API consistency for `credentials_provider` and `credentials_cache` in other builders
+
+Keeping the credentials caching code in the `aws-config` crate, we considered replacing the `credentials_provider` method with `credentials_cache` method in  `aws_types::sdk_config::Builder` and service clients' `config::Builder`s. This will help constrain the use of the APIs in that `ConfigLoader` is the only place that can configure a credentials provider and a credentials cache; `sdk_config::Builder` and service client's `config::Builder` can only receive a `SharedCredentialsCache` already constructed by `ConfigLoader`. But the constraint is a bit too much because those who directly interact with `sdk_config::Builder` and `config::Builder` cannot create a credentials cache, making those builders pretty much unusable.
+
+### Creating credentials caches outside the `aws-config` crate
+
+There have been no rejected alternatives discovered because the code for credentials caching needs to go somewhere outside the `aws-config` crate anyway. Existing crate dependencies act as constraints and have forced us to place credentials caching and providers as described in the proposed solution.
+
+Changes Checklist
+-----------------
+The following list does not repeat what is listed in the preceding RFC but does include those new mentioned in the `Assumptions` section:
+
+- [ ] Create and name the `aws-XXX` crate
```

Comment:
Seems like it should be something along the lines of `aws-credentials`, although I'm worried that name will imply it holds all the default credential provider implementations, when really it only holds some traits and cache impls.

Reply:
Hmm, sounds like there could be a mental model related problem but what would be a practical problem if the default credential provider implementation went into this new crate?

Reply:
I'll circle back and update the RFC as I refine what really needs to go into this new crate.

Reply:
Updated the selected design as per our discussion and named the crate `aws-credential-types` in fb28d8c.